### PR TITLE
open-mpi: Set --HEAD to make release build

### PR DESCRIPTION
**Pull request by aw1621107**
open-mpi from git configs a debug build by default. See HACKING in the
repo root for details

### CODE REVIEW

**Comment by DomT4** [comment]
Is this something we should only be passing when the `head` build is requested or?

**Comment by aw1621107** [comment]
Whoops, you're right. Fixing...

File: Library/Formula/open-mpi.rb [hunk 1]
```diff
@@ -41,6 +41,7 @@ def install
       --enable-ipv6
       --with-libevent=#{Formula["libevent"].opt_prefix}
       --with-sge
+      --with-platform=optimized if build.head?
```
Comment (dunn) [inline Library/Formula/open-mpi.rb]: This is adding `--with-platform-optimized`, `if`, and `build.head?` as three separate arguments.  Instead add this below:
```ruby
args << `--with-platform-optimized` if build.head?

Comment (aw1621107) [inline Library/Formula/open-mpi.rb]: Whoops, didn't realize that. Thanks!

**Comment by dunn** [comment]
Thanks! Merged in 96fe7d1818822f497de21ea5a98ad3c271bcbbb0.

**Pull request closed without merging**